src/viam/sdk/registry/registry.hpp

```diff
@@ -169,11 +166,24 @@ class Registry {
     /// @return A `shared_ptr` to the registered api's `ResourceRegistration`.
     static std::shared_ptr<ResourceRegistration> lookup_resource(API api);
 
-    /// @brief Provide information on registered resources.
-    /// @return A map from name to `ModelRegistration` of all registered resources.
-    static std::unordered_map<std::string, std::shared_ptr<ModelRegistration>> registered_models();
+    /// @brief Provide information on registered resource models.
+    /// @return A map from name to `ModelRegistration` of all registered resource models.
+    static const std::unordered_map<std::string, std::shared_ptr<ModelRegistration>>&
```

Comment:
I have some concerns about these return types. First, it is returning a reference to internal state. Are there concurrency concerns? Second, `shared_ptr` is const, but the `ModelRegistration` it points to is not.

Reply:
Ahh good shout out. I think this overlaps with your concerns on the `resource_manager` method on `ResourceServer`. I think some revisiting of why we need to expose these types at all, how they're being used, and how to avoid doing so is warranted. I'll do some digging/refactoring.